compile.js

```diff
@@ -192,8 +316,8 @@ function compileFieldWrite(ctx, field, name) {
     }
 }
 
-function compileMapRead(readCode, name, numRepeated) {
-    return (numRepeated ? '' : 'var ') + 'entry = ' + readCode + '; obj.' + name + '[entry.key] = entry.value';
+function compileMapRead(readCode, name) {
+    return 'entry = ' + readCode + '; obj.' + name + '[entry.key] = entry.value';
 }
 
 function compileRepeatedWrite(ctx, field, numRepeated) {
```

Comment:
Other typescript issue:
If I have a proto containing:
```
repeated customtype val = 1;
map<string, int32> map = 2;
```
I will get something like:
`if (obj.val&& pbf) for (var i = 0;  ` ...
`if (obj.map && pbf) for (i in ` ...
The type of i is an number, but later it becomes a string.

A way to solve this is to do something like:
```
function compileRepeatedWrite(ctx, field, numRepeated) {
  var iVal = 'i' + (numRepeated ? numRepeated.toString() : '');
  return (
    'for (var ' +
    iVal +
    ' = 0; ' +
    iVal +
    ' < obj.' +
    field.name +
    '.length; ' +
    iVal +
    '++) ' +
    compileFieldWrite(ctx, field, 'obj.' + field.name + '[' + iVal + ']')
  );
}

function compileMapWrite(ctx, field, numRepeated) {
  var name = 'obj.' + field.name;
  var iVal = 'i' + (numRepeated ? numRepeated.toString() : '');

  return (
    'for (var ' +
    iVal +
    ' in ' +
    name +
    ') if (Object.prototype.hasOwnProperty.call(' +
    name +
    ', ' +
    iVal +
    ')) ' +
    compileFieldWrite(
      ctx,
      field,
      '{ key: ' + iVal + ', value: ' + name + '[' + iVal + '] }'
    )
  );
}
```